resources/lib/functions.py

```diff
@@ -585,12 +585,16 @@ def processDirectory(results, progress, params):
     name_format = params.get("name_format", None)
     if name_format is not None:
         name_format = urllib.unquote(name_format)
-        name_format = settings.getSetting(name_format)
-
+        name_formatting = settings.getSetting(name_format)
+        if name_format == "latest_name_format":
+            name_formatting = settings.getSetting("episode_name_format")
     dirItems = []
-    result = results.get("Items")
-    if result is None:
+    if results is None:
         result = []
+    if "Items" in results:
+        result = results.get("Items") 
```

Comment:
oh I get it now, its because it is a list instead of an dict.
Try this:
```
if isinstance(results, dict):
   result = results.get("Items") 
else:
   result = results
```
Sorry to be stuck on this but it is all about intent of the code, think about this, if you read this code in 12 months what is the code trying to do. In the above snip it is easy to see that the response result can be either a list or an dict and you do something different for each. You can see that right there in the code, it is self explanatory.